### PR TITLE
[WIP] Fix StrongNameTests

### PR DESCRIPTION
**Pull request by RassK**
## What

Tries to fix the StrongNameTests by adding some time to exit.

## Tests

Existing.

### CODE REVIEW

File: test/test-applications/integrations/TestApplication.StrongNamed/Program.cs [hunk 1]
```diff
@@ -27,5 +27,18 @@ public static void Main(string[] args)
...
+        // The "LONG_RUNNING" environment variable is used by tests that access/receive
+        // data that takes time to be produced.
+        var longRunning = Environment.GetEnvironmentVariable("LONG_RUNNING");
+        while (longRunning == "true")
+        {
+            // In this case it is necessary to ensure that the test has a chance to read the
+            // expected data, only by keeping the application alive for some time that can
+            // be ensured. Anyway, tests that set "LONG_RUNNING" env var to true are expected
+            // to kill the process directly.
+            Console.WriteLine("LONG_RUNNING is true, waiting for process to be killed...");
```
Comment (pellared) [inline test/test-applications/integrations/TestApplication.StrongNamed/Program.cs:40]: OutOfMemory + this will never finish 😉

Comment (Kielek) [reply test/test-applications/integrations/TestApplication.StrongNamed/Program.cs:38]: Please use: https://github.com/Kielek/opentelemetry-dotnet-instrumentation/blob/4fc8a4a76371e9936863cda273e38f8d55f2407c/test/test-applications/integrations/TestApplication.Smoke/Program.cs#L40-L48

Comment (Kielek) [reply test/test-applications/integrations/TestApplication.StrongNamed/Program.cs:39]: On CI, something is actively writing to the standard input so it produces tons of logs.

Comment (pellared) [reply test/test-applications/integrations/TestApplication.StrongNamed/Program.cs:39]: and the test would also have to kill the app...

File: test/IntegrationTests/StrongNamedTests.cs [hunk 2]
```diff
@@ -36,6 +36,8 @@ public StrongNamedTests(ITestOutputHelper output)
     [Fact]
     public async Task SubmitsTraces()
     {
+        SetEnvironmentVariable("LONG_RUNNING", "true");
+
```
Comment (Kielek) [inline test/IntegrationTests/StrongNamedTests.cs:40]: later in this file you need to manually kill the application.

**Review by pjanotti — APPROVED** [review]
LGTM. Could you please also add similar changes to Program.cs for the DomainNeutralTests? Although it is not failing (afaik) it seems reasonable for it to also be a "LONG_RUNNING" one too.

**Comment by pjanotti** [comment]
@RassK in light of https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/1242#issuecomment-1254480867 I don't think this will solve the issue.

**Comment by RassK** [comment]
> @RassK in light of [#1242 (comment)](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/1242#issuecomment-1254480867) I don't think this will solve the issue.

Yep, after I saw 0 received, then I already thought, probably not gonna solve.

**Pull request closed without merging**